nest/src/command/install.rs

```diff
@@ -1,123 +1,84 @@
 use libnest::system::System;
-use progressbar::ProgressBar;
 
+use command::download::download_package;
+use progress::Progress;
+use progressbar::ProgressBar;
 use query;
 
-/// Installs all the given packages
+/// Installs the given package
 ///
-/// This will go through all targets, checks that the package exists, resolves the dependency graph,
-/// ensures the installation will not break anything or delete any file and installs the package.
-pub fn install(config: &Config, matches: &ArgMatches) {
-    let sys = System::current();
-    let mut targets = Vec::new();
+/// This function will draw a progress bar on the user's output
+/// It will return `Ok` if the install succeed, or `Err` otherwise.
+pub fn install_package(
+    config: &Config,
+    sys: &System,
+    progress: &Progress,
+    target: &Package,
+) -> Result<(), Error> {
+    let mut old_state = InstallState::Waiting;
+    let mut pb = ProgressBar::new(old_state.to_string());
+    pb.set_target(format!(
+        "({}) {}",
+        progress,
+        target.manifest().metadatas().name()
+    ));
 
-    // Search in the cache for manifests, ensuring the target exists.
-    for target in matches.values_of("PACKAGE").unwrap() {
-        match query::cache(config, target) {
-            Some(query) => {
-                match query.perform() {
-                    Ok(mut manifests) => {
-                        match manifests.len() {
-                            0 => {
-                                eprintln!("{}: couldn't find a package with name \"{}\". Try \"{}\" to look for an existing package.",
-                                    red!("error"),
-                                    purple!(target),
-                                    purple!("nest search")
-                                );
-                                process::exit(1);
-                            }
-                            1 => targets.append(&mut manifests),
-                            len => {
-                                eprintln!("{}: found {} packages with name \"{}\". Please be more explicit.", red!("error"), len, purple!(target));
-                                for (i, package) in manifests.iter().enumerate() {
-                                    eprintln!("\t- {}", package);
-                                    if i == 9 && len > 10 {
-                                        eprintln!("and {} more...", len - 10);
-                                        break;
-                                    }
-                                }
-                                process::exit(1);
-                            }
-                        }
-                    }
-                    Err(e) => {
-                        eprintln!("{}: couldn't go through cache: {}.", red!("error"), e);
-                        process::exit(1);
-                    }
-                }
+    let r = sys.installer(&target.data_path(config), &target.manifest())
```

Comment:
`r` could be named better.

Reply:
Actually, it's r for `Result<T, U>`, like people tend to use `s` for `Strings`, but yeah, i understand.

Reply:
If you remember, I used to use single letter variables all the time, but I'm really against it now: searching for them tends to produce lot of false positives (even if you limit yourself to a bloc of code) and I actually find variables less then 3 letters long easy to "miss" in function parameter lists, constructors etc.